internal/controller/ols_app_redis_assets.go

```diff
@@ -11,14 +17,52 @@ import (
 	olsv1alpha1 "github.com/openshift/lightspeed-operator/api/v1alpha1"
 )
 
+var DeploymentSelectorLabels = map[string]string{
+	"app.kubernetes.io/component":  "redis-server",
+	"app.kubernetes.io/managed-by": "lightspeed-operator",
+	"app.kubernetes.io/name":       "lightspeed-service-redis",
+	"app.kubernetes.io/part-of":    "openshift-lightspeed",
+}
+
 func (r *OLSConfigReconciler) generateRedisDeployment(cr *olsv1alpha1.OLSConfig) (*appsv1.Deployment, error) {
 	cacheReplicas := int32(1)
-	DeploymentSelectorLabels := map[string]string{
-		"app.kubernetes.io/component":  "redis-server",
-		"app.kubernetes.io/managed-by": "lightspeed-operator",
-		"app.kubernetes.io/name":       "lightspeed-service-redis",
-		"app.kubernetes.io/part-of":    "openshift-lightspeed",
+	revisionHistoryLimit := int32(0)
+	redisPassword, err := getSecretContent(r.Client, cr.Spec.OLSConfig.ConversationCache.Redis.CredentialsSecretRef.Name, cr.Namespace, OLSPasswordFileName)
+	if err != nil {
+		return nil, fmt.Errorf("Password is a must to start redis deployment : %w", err)
+	}
+	volumes := []corev1.Volume{}
+	tlsCertsVolume := corev1.Volume{
+		Name: "secret-" + OLSAppRedisCertsSecretName,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: OLSAppRedisCertsSecretName,
+			},
+		},
+	}
+	redisCAConfigVolume := corev1.Volume{
+		Name: OLSRedisCAVolumeName,
+		VolumeSource: corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: OLSRedisCACmName,
+				},
+			},
+		},
 	}
+	volumes = append(volumes, tlsCertsVolume, redisCAConfigVolume)
+	volumeMounts := []corev1.VolumeMount{}
+	redisTLSVolumeMount := corev1.VolumeMount{
+		Name:      "secret-" + OLSAppRedisCertsSecretName,
+		MountPath: OLSAppCertsMountRoot,
+		ReadOnly:  true,
+	}
+	redisCAVolumeMount := corev1.VolumeMount{
+		Name:      OLSRedisCAVolumeName,
+		MountPath: path.Join(OLSAppCertsMountRoot, OLSRedisCAVolumeName),
+		ReadOnly:  true,
+	}
+	volumeMounts = append(volumeMounts, redisTLSVolumeMount, redisCAVolumeMount)
```

Comment:
as the only append happens here, I think this can be more simple:
```go
volumeMounts := []corev1.Volume{redisTLSVolumeMount,redisCAVolumeMount}
```

Reply:
Updated. Thanks